### ISSUE TITLE
typeof AND operator

### ISSUE THREAD

**Issue by AmmarHasan** [microsoft/TypeScript#43946]
# Bug Report

The logical AND operator does not seem to work if both the operands has `typeof` operator.

### 🔎 Search Terms

typeof AND operator

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/DYUwLgBAhgcgrgWwEYgE4QFwQHaJegXggEYBuAKFEigGUxUBLbAc0wgGd6nmLyGAzCAAowATwAOIAPaDYeNAUUByToxZKIAMk0QxkmdHjIFy3MdRKAlBADe5CNDprWROefIBfIA)

### 💻 Code

```ts
let aNumber : number = 1;
let aString : string;

if (typeof aNumber === 'string' && typeof aNumber === 'number') {
  aString=aNumber
}
```

### 🙁 Actual behavior

Does not throw any error on the IF condition

### 🙂 Expected behavior

Should throw an error because `aNumber` cannot be of type `number` and `string` at the same time

**Comment by RyanCavanaugh** [comment]
The result type of `typeof` doesn't change based on the operand; this is the intended behavior.

There's no error on the assignment because the code is unreachable; this is also the intended behavior.
